Symfony/CS/Console/ConfigurationResolver.php

```diff
@@ -350,8 +355,8 @@ private function resolveConfig()
      */
     private function resolveConfigPath()
     {
-        if (is_file($this->path)) {
-            $this->config->finder(new \ArrayIterator(array(new \SplFileInfo($this->path))));
+        if (count($this->path) === 1 && is_file($this->path[0])) {
+            $this->config->finder(new \ArrayIterator(array(new \SplFileInfo($this->path[0]))));
         } elseif ($this->isStdIn) {
             $this->config->finder(new \ArrayIterator(array(new StdinFileInfo())));
         } elseif (null !== $this->path) {
```

Comment:
Here is your problem.
When path is array then `$this->config->setDir($this->path);` is called, and when one of item inside array is not a dir but file - it will fail.

But this just show bigger issue with your handling of config files.
Here, in ConfigurationResolver, we resolve Config instance.

In `computeConfigFiles` you create _candidates_ for every passed path to be fixed. But then, in `resolveConfig` only the first candidate that was found is taken. That way only the first config will be used - for every passed path that need fixing.

Then, here, in `resolveConfigPath` we set a finder/dir (dir is only a shorthand for setting a finder) inside config. And if one want to fix multiple paths  they are just set all to one config instance.

All thing is about a config instance. For now - we have one instance. If we want to handle with many paths to be fixed then we need to handle many configs. And current implementation of `FixCommand` simply don't handle it.